pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ClusterMigrationTest.java

```diff
@@ -458,8 +458,8 @@ public void testClusterMigrationWithReplicationBacklog(boolean persistent, Subsc
         retryStrategically((test) -> !topic1.isReplicationBacklogExist(), 10, 1000);
         assertFalse(topic1.isReplicationBacklogExist());
 
-        producer1.send("test".getBytes());
         // verify that the producer1 is now connected to migrated cluster "r2" since backlog is cleared.
+        retryStrategically((test) -> topic2.getProducers().size() == 2, 60, 1000);
```

Comment:
So we can remove line-463

Reply:
No. Line 463 is assertion. Line 462 just does retries until the condition satisfies and exits after the specified retries are exhausted.